src/platform/plugins/shared/dashboard/public/dashboard_api/layout_manager/layout_manager.ts

```diff
@@ -490,17 +528,21 @@ export function initializeLayoutManager(
         const controlToPin = layout$.getValue().panels[uuid];
         if (!controlToPin) return;
 
-        // add control panel to the end of the pinned controls
-        const newControls = { ...layout$.getValue().controls };
-        newControls[uuid] = {
-          type: controlToPin.type as StickyControlLayoutState['type'],
-          order: Object.keys(newControls).length,
+        pinPanel(uuid, controlToPin);
+      },
+      addPinnedPanel: async (panelPackage: PanelPackage) => {
+        const newPanelUuid = createPanel(panelPackage);
+        const { serializedState } = panelPackage;
+        const displaySettings = serializedState
+          ? pick(serializedState.rawState, 'grow', 'width')
+          : {};
+        const panelToPin = {
+          type: panelPackage.panelType,
+          grid: { x: 0, y: 0, w: 1, h: 1 },
```

Comment:
Why are we including grid information here? Panels in the pinned section shouldn't have grid information

Reply:
Required by `DashboardLayoutPanel` which is the type sig we get from `layout$.panels`, though I guess we can type it differently now that we're abstracting `pinPanel` out

Reply:
Yeah, let's be more clever with typing in this case instead of adding garbage data I think :)